Sources/BraveWallet/Crypto/Transaction Confirmations/TransactionStatusView.swift

```diff
@@ -0,0 +1,181 @@
+            .padding(.top, 10)
+          Text(confirmationStore.activeTxStatus == .confirmed ? Strings.Wallet.confirmedTransactionDescription : Strings.Wallet.failedTransactionDescription)
+            .font(.subheadline)
+            .foregroundColor(Color(.secondaryBraveLabel))
+            .multilineTextAlignment(.center)
+          if confirmationStore.activeTxStatus == .error, let txProviderError = confirmationStore.transactionProviderErrorRegistry[confirmationStore.activeTransactionId] {
+            StaticTextView(text: "\(txProviderError.code): \(txProviderError.message)")
+              .frame(maxWidth: .infinity)
+              .frame(height: 100)
+              .background(Color(.tertiaryBraveGroupedBackground))
+              .clipShape(RoundedRectangle(cornerRadius: 5, style: .continuous))
+              .padding()
+              .background(
+                Color(.secondaryBraveGroupedBackground)
+              )
+              .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+              .padding(.top, 10)
+          }
+          HStack {
+            if confirmationStore.activeTxStatus == .confirmed {
+              Button {
+                transactionDetails = confirmationStore.activeTxDetailsStore()
+              } label: {
+                Text(Strings.Wallet.confirmedTransactionReceiptButtonTitle)
+              }
+              .buttonStyle(BraveOutlineButtonStyle(size: .large))
+            }
+            Button {
+              onDismiss()
+            } label: {
+              Text(Strings.Wallet.confirmedTransactionCloseButtonTitle)
+            }
+            .buttonStyle(BraveFilledButtonStyle(size: .large))
+          }
+          .padding(.top, 40)
+        }
+        .frame(maxWidth: .infinity)
+        .frame(minHeight: geometry.size.height)
+        .padding()
+      }
+      .background(
+        Color(.braveGroupedBackground)
+          .edgesIgnoringSafeArea(.all)
+      )
+    }
+  }
+  
+  var body: some View {
+    if confirmationStore.isTxSubmitting {
+      loadingTxView
+    } else {
+      switch confirmationStore.activeTxStatus {
+      case .signed, .submitted:
+        signedOrSubmittedTxView
+      case .confirmed, .error:
+        if confirmationStore.activeTxStatus == .error, confirmationStore.activeTransactionId == "" {
+          EmptyView()
+        } else {
+          confirmedOrFailedTxView
+        }
+      default:
+        EmptyView()
+      }
+    }
+  }
+}
+
+/// A custom loader needed during the process of submitting a transaction
+private struct TxProgressView: View {
+  @State private var isAnimating = false
+  
+  var body: some View {
+    Image("tx-loading", bundle: .module)
+      .rotationEffect(.degrees(isAnimating ? 360 : 0))
+      .animation(isAnimating ? .linear(duration: 1.0).repeatForever(autoreverses: false) : .default)
+      .onAppear { isAnimating = true }
+      .onDisappear { isAnimating = false }
+  }
+}
```

Comment:
Can we not just use `BraveCircularProgressViewStyle`?

Reply:
updated in [02e39dd](https://github.com/brave/brave-ios/pull/6960/commits/02e39dd3500416bf27c899accce6252b59e78d3f)